scala2java-core/src/main/scala/io/github/effiban/scala2java/core/predicates/CoreInvocationArgByNamePredicate.scala

```diff
@@ -11,7 +11,9 @@ object CoreInvocationArgByNamePredicate extends InvocationArgByNamePredicate {
 
   private final val MethodsWithFirstArgPassedByName: Set[Term] = Set(
     Term.Select(Term.Name(TermNameValues.Try), Term.Name(Apply)),
-    Term.Select(Term.Name(TermNameValues.Future), Term.Name(Apply))
+    Term.Select(Term.Name(TermNameValues.Try), Term.Name(JavaOfSupplier)),
+    Term.Select(Term.Name(TermNameValues.Future), Term.Name(Apply)),
+    Term.Select(Term.Name(TermNameValues.JavaCompletableFuture), Term.Name(JavaSupplyAsync))
```

Comment:
This predicate was initially supposed to be applied to Java-style method invocations only.|
However only now the desugaring is properly supported for `Term.Select`, so that the Java-style method invocations will now be passed in here for those cases, and they need to be added.
The Scala-style invocations must remaiun until support for desugaring of `Term.ApplyType` is supported.